src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -8299,7 +8199,7 @@ bool StorageReplicatedMergeTree::removeSharedDetachedPart(DiskPtr disk, const St
             String id = disk->getUniqueId(checksums);
             bool can_remove = false;
             std::tie(can_remove, files_not_to_remove) = StorageReplicatedMergeTree::unlockSharedDataByID(id, table_uuid, part_name,
-                detached_replica_name, disk, zookeeper, getContext()->getReplicatedMergeTreeSettings(), log,
+                detached_replica_name, disk, zookeeper, local_context->getReplicatedMergeTreeSettings(), &Poco::Logger::get("StorageReplicatedMergeTree"),
```

Comment:
Why `log` is replaced with `&Poco::Logger::get("StorageReplicatedMergeTree")`?

Reply:
Because method became static

Reply:
It might make sense to pass logger through function arguments, but it already has too many arguments, so maybe it's not that bad to use `&Poco::Logger::get("StorageReplicatedMergeTree")`